compose/cli/main.py

```diff
@@ -257,7 +257,7 @@ def build(self, options):
         e.g. `composetest_db`. If you change a service's `Dockerfile` or the
         contents of its build directory, you can run `docker-compose build` to rebuild it.
 
-        Usage: build [options] [--build-arg key=val...] [SERVICE...]
+        Usage: build [options] [--] [--build-arg key=val...] [SERVICE...]
```

Comment:
this doesn't look correct, as it would mean `--` can be before `--build-arg` and this means build args would not be processed as option?

Reply:
Maybe I misunderstood `--build-arg`, but as part of the POSIX standard, I believe, `--`  is intended to signify no flags are present. 

For example with `rm` it can be used on files with `-` in the name, e.g, `rm -- -foo`

Reply:
It means "stop processing" after,
so I could do (silly example);

```
docker-compose build --pull -- --pull
```

to build a service named `--pull` (but also have the `--pull` option set